clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1088,14 +1090,16 @@ public void run() {
                             markCoordinatorUnknown();
                         } else if (heartbeat.pollTimeoutExpired(now)) {
                             // the poll timeout has expired, which means that the foreground thread has stalled
-                            // in between calls to poll(), so we explicitly leave the group.
-                            log.warn("This member will leave the group because consumer poll timeout has expired. This " +
-                                    "means the time between subsequent calls to poll() was longer than the configured " +
-                                    "max.poll.interval.ms, which typically implies that the poll loop is spending too " +
-                                    "much time processing messages. You can address this either by increasing " +
-                                    "max.poll.interval.ms or by reducing the maximum size of batches returned in poll() " +
-                                    "with max.poll.records.");
-                            maybeLeaveGroup();
+                            // in between calls to poll().
+                            if (isDynamicMember()) {
+                                log.warn("This member is at the risk of leaving the group because consumer poll timeout has expired. This " +
+                                        "means the time between subsequent calls to poll() was longer than the configured " +
+                                        "max.poll.interval.ms, which typically implies that the poll loop is spending too " +
+                                        "much time processing messages. You can address this either by increasing " +
+                                        "max.poll.interval.ms or by reducing the maximum size of batches returned in poll() " +
+                                        "with max.poll.records.");
+                            }
+                            maybeLeaveGroup("the consumer is lagging");
```

Comment:
I was thinking that we can just pass in the statement in the above warn as the root cause into `maybeLeaveGroup`.